architecture-decisions/0002-shibboleth.md

```diff
@@ -0,0 +1,19 @@
+# 2. Use Shibboleth for authentication
+
+Date: 2023-10-12
+
+## Status
+
+Discussion
+
+## Context
+
+Users must be able to sign into the TigerData Rails application in such a way that they are also securely authenticated to MediaFlux, and actions that they then take in MediaFlux are accurately tied to their user.  Theoretically, this could be accomplished by minting an authentication token in MediaFlux based on the value of a SAML 2.0 cookie created by the Shibboleth session when a user logs into the Rails application via CAS tied to a Shibboleth service that runs on the MediaFlux server.  For the sake of security, the Shibboleth service should run on the MediaFlux server.  Our Rails application will interact Shibboleth for authentication using CAS.  Note that the work to tie a session cookie to a MediaFlux token is theoretical and needs to be tested to confirm whether or not it will work for tying the user in Rails to a MediaFlux session.
```

Comment:
I am sorry, should this be `will interact with Shibboleth for authentication` please?

Reply:
Done.